src/java.base/share/classes/com/sun/crypto/provider/AESCrypt.java

```diff
@@ -1365,8 +1365,15 @@ private void makeSessionKey(byte[] k) throws InvalidKeyException {
         int ROUND_KEY_COUNT = (ROUNDS + 1) * 4;
 
         int BC = 4;
-        int[][] Ke = new int[ROUNDS + 1][4]; // encryption round keys
-        int[][] Kd = new int[ROUNDS + 1][4]; // decryption round keys
+        int[][] Ke = new int[ROUNDS + 1][]; // encryption round keys
+        int[][] Kd = new int[ROUNDS + 1][]; // decryption round keys
```

Comment:
This is more a question of curiosity and likely beyond the scope of this PR/bug ID -- would it be beneficial to refactor this to allocate and operate over just two arrays, one for Ke and one for Kd, effectively removing the need for allocation in `expandToSubKey` as well?

```
int[] Ke = new int[4 * (ROUNDS + 1)];
int[] Kd = new int[4 * (ROUNDS + 1)];
```

Reply:
True, let me try that!

Reply:
New commit implements this, with even more performance benefits.

Reply:
Excellent, thanks!